contracts/data-storage/andromeda-boolean/src/query.rs

```diff
@@ -0,0 +1,29 @@
+use crate::state::{DATA, DATA_OWNER, RESTRICTION};
+use andromeda_data_storage::boolean::{BooleanRestriction, GetDataOwnerResponse, GetValueResponse};
+use andromeda_std::{ado_contract::ADOContract, amp::AndrAddr, error::ContractError};
+use cosmwasm_std::{Addr, Storage};
+
+pub fn has_permission(storage: &dyn Storage, addr: &Addr) -> Result<bool, ContractError> {
```

Comment:
What's the difference between this and our permissioning? I'd understand if it was key specific but I think this is covered by default ADO functionality

Reply:
If the type of restriction is `Restricted`, only the owner of stored data has permission to set or delete.
It is similar to key specific, but just permission for data without a key. This is why the `DATA_OWNER` is set.
@crnbarr93

Reply:
Say one user sets the value when the restriction is `public`, and then if the restriction is set as `Restricted`, only `DATA_OWNER` and owner/operator of the ADO can set or delete the stored value.

Reply:
Ok but our permissioning system can be used to just restrict access to `set_value` which has the same outcome, no?

Reply:
I resolved it.